python/arcticdb/version_store/_normalization.py

```diff
@@ -497,6 +514,20 @@ def _index_to_records(self, df, pd_norm, dynamic_strings, string_max_len):
             df.reset_index(fields, inplace=True)
             index = df.index
         else:
+            n_rows = len(index)
+            n_categorical_columns = len(df.select_dtypes(include="category").columns)
+            if IS_PANDAS_TWO and isinstance(index, (Index, RangeIndex)) and n_rows == 0 and n_categorical_columns == 0:
+                # In Pandas 1.0, an Index is used by default for any an empty dataframe or series is created, except if
+                # there are categorical columns in which case a RangeIndex is used.
+                #
+                # In Pandas 2.0, RangeIndex is used by default for _any_ an empty dataframe or series is created.
+                # See: https://github.com/pandas-dev/pandas/issues/49572
+                # Yet internally, ArcticDB uses a DatetimeIndex for empty dataframes and series without categorical
+                # columns.
+                #
+                # The index is converted to a DatetimeIndex for preserving the behavior of ArcticDB with Pandas 1.0.
+                index = DatetimeIndex([])
```

Comment:
I came up with two tests for safe normalisation:

* The data frame we read back should be equivalent to the input
* If the user had a particular (index) type in mind but accidentally generated a zero-row data frame initially, the user should be able to append to such a frame

So the best solution is to introduce something in the C++ to allow the index type of an empty DF:

* to be accurately saved
* to change on the first append